docs/content/1.guide/1.configuration.md

```diff
@@ -40,81 +38,82 @@ nitro.timing=true
 
-**Example:**
+First, you need to define the runtime config in your configuration file.
 
 ::code-group
 ```ts [nitro.config.ts]
 export default defineNitroConfig({
   runtimeConfig: {
-    helloThere: "foobar",
+    apiToken: "dev_token", // `dev_token` is the default value
   }
 })
 ```
 
 ```ts [nuxt.config.ts]
 export default defineNuxtConfig({
   runtimeConfig: {
-    helloThere: "foobar",
+    apiToken: "dev_token", // `dev_token` is the default value
   }
 })
 ```
 ::
 
-You can now access the runtime config using `useRuntimeConfig(event)`.
-
-**Example:**
+You can now access the runtime config using `useRuntimeConfig(event)`.  Use `useRuntimeConfig(event)` within event handlers and utilities and **avoid** calling it in ambient global contexts. This could lead to unexpected behavior such sharing the same runtime config across different requests.
 
 ::code-group
 ```ts [api/example.get.ts (nitro)]
 export default defineEventHandler((event) => {
-  return useRuntimeConfig(event).helloThere // foobar
+  return useRuntimeConfig(event).apiToken // Returns `dev_token`
 });
 ```
 
 ```ts [server/api/example.get.ts (nuxt)]
 export default defineEventHandler((event) => {
-  return useRuntimeConfig(event).helloThere // foobar
+  return useRuntimeConfig(event).apiToken // dev_token
 });
 ```
 ::
 
-::alert{type="info"}
-**Note:** Consider using `useRuntimeConfig(event)` within event handlers and utilities and avoid calling it in ambient global contexts.
-::
-
+`useRuntimeConfig(event)` is an auto-imported function. Read more about [auto-imports](/auto-imports).
 
-### Environment Variables
+### Local development
 
-Nitro supports defining environment variables using `.env` file in development (use platform variables for production).
+Finally, you can update the runtime config using environment variables. You can use a `.env` file in development and use platform variables in production (see below).
 
 Create an `.env` file in your project root:
 
 ```bash [.env]
-TEST="123"
+API_TOKEN="123"
 ```
 
-You can universally access environment variables using `import.meta.env.TEST` or `process.env.TEST`.
-
-::alert{type="info"}
-**Note:** Consider writing any logic that depends on environment variables within event handlers and utilities and avoid accessing and caching them in ambient global contexts.
-::
+Re-start the development server, fetch the `/api/example` endpoint and you should see `123` as the response instead of `dev_token`.
 
-### Update runtime config using environment variables
+Do not froget taht you can still universally access environment variables using `import.meta.env` or `process.env` but avoid using them in ambiant global contexts to prevent unexpected behavior.
```

Comment:
```suggestion
Do not forget that you can still universally access environment variables using `import.meta.env` or `process.env` but avoid using them in ambiant global contexts to prevent unexpected behavior.
```